lib/auth/auth.go

```diff
@@ -634,6 +697,14 @@ func (a *Server) generateUserCert(req certRequest) (*certs, error) {
 			ClusterName: req.appClusterName,
 		},
 		TeleportCluster: clusterName,
+		RouteToDatabase: tlsca.RouteToDatabase{
+			ServiceName: req.dbService,
+			Protocol:    req.dbProtocol,
+			Username:    req.dbUser,
+			Database:    req.dbName,
```

Comment:
do we validate that `req.dbUser` and `req.dbName` are within the allowed users/databases (`dbNames/dbUsers`)?

Reply:
I though about this as well, but I don't think we can do it here because we'd need to check them against db names/users of the mapped role in case of trusted clusters which we don't have here.

Reply:
makes sense
other possible places to authorize this:
- lib/srv/db/server.go (`authorize` method)
- lib/auth/permissions.go (`authorizeRemoteUser`/`authorizeLocalUser` methods)

Reply:
Yeah, db user/name that a user uses to connect to the database are authorized by the engine's `checkAccess` method which calls checker's `CheckAccessToDatabase`, if I understand correctly what you mean. So if a user tries to use a name/user that's not allowed by their role (mapped, in case of tc), they will get access denied.